src/differentiation/compute_jacobian_ad.jl

```diff
@@ -44,7 +44,7 @@ function ForwardColorJacCache(f::F, x, _chunksize = nothing;
         p = adapt.(parameterless_type(x), generate_chunked_partials(x, colorvec, chunksize))
         _t = Dual{T, eltype(x), getsize(chunksize)
                   }.(vec(x), ForwardDiff.Partials.(first(p)))
-        t = ArrayInterface.restructure(x, _t)
+        t = ArrayInterface.restructure(similar(x, eltype(_t)), _t)
```

Comment:
A hack that gets things to work for now. I don't quite understand the semantics of `restructure`, but if it should automatically convert the type for us then this shouldn't be necessary in theory: see https://github.com/JuliaArrays/ArrayInterface.jl/issues/404. The `similar` can also cause an extra allocation in cases.